src/extension.ts

```diff
@@ -838,11 +838,12 @@ function launchMetals(
       decorationType = window.createTextEditorDecorationType(options);
     });
     client.onNotification(DecorationsRangesDidChange.type, (params) => {
-      const editor = window.activeTextEditor;
-      if (
-        editor &&
-        Uri.parse(params.uri).toString() === editor.document.uri.toString()
-      ) {
+      const editors = window.visibleTextEditors;
+      const path = Uri.parse(params.uri).toString();
+      const editor = editors.find(
+        (editor) => editor.document.uri.toString() == path
+      );
+      if (editors.length != 0 && editor) {
```

Comment:
Why the length check?

Reply:
och right, it's actually not needed. Thanks!

Reply:
Fixed!